archai/discrete_search/api/model_evaluator.py

```diff
@@ -131,3 +131,31 @@ def fetch_all(self) -> List[Optional[float]]:
         """
 
         pass
+
+
+class ProgressiveModelEvaluator(EnforceOverrides):
+    """Abstract class for progressive model evaluators. Progressive evaluators are stateful evaluators
+    that can be saved and loaded from checkpoints. This is used by search algorithms like `ProgressiveHalving`
+    to save the state of the evaluator and restore it later.
+
+    Subclasses of `ProgressiveModelEvaluator` are expected to implement `ProgressiveModelEvaluator.save_checkpoint`
+    and `ProgressiveModelEvaluator.load_checkpoint`, and should also be subclasses of `ModelEvaluator`/`AsyncModelEvaluator`.
```

Comment:
can't be both right? Perhaps should read "must be a subclass of either `ModelEvaluator` or `AsyncModelEvaluator`" ?

Reply:
Good point. Fixed